image_tools/src/showimage_main.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include "image_tools/options.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "image_tools/showimage.hpp"
+#include "image_tools/cam2image.hpp"
+
+
+int main(int argc, char * argv[])
+{
+  // Force flush of the stdout buffer.
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+
+  // Initialize any global resources needed by the middleware and the client library.
+  // This will also parse command line arguments one day (as of Beta 1 they are not used).
+  // You must call this before using any other part of the ROS system.
+  // This should be called once per process.
+  rclcpp::init(argc, argv);
+
+  // Create an executor that will be responsible for execution of callbacks for a set of nodes.
+  // With this version, all callbacks will be called from within this thread (the main one).
+  rclcpp::executors::SingleThreadedExecutor exec;
+  rclcpp::NodeOptions options;
+
+  // Add some nodes to the executor which provide work for the executor during its "spin" function.
+  // An example of available work is executing a subscription callback, or a timer callback.
+
+  auto showimage = std::make_shared<image_tools::ShowImage>(options);
+  if (showimage->setup(argc, argv)){
+    exec.add_node(showimage);
+    showimage->execute();
```

Comment:
Currently, this doesn't work if I want to load into a container dynamically, e.g.

```sh
ros2 component standalone image_tools image_tools::ShowImage
```

What if we called the execute method in the constructor (or as a callback of a oneshot timer started in the constructor)? Rather than having a setup method, we could overload the constructor for the purpose of keeping the demo backwards compatible, i.e.

```c++
ShowImage(rclcpp::NodeOptions options);
ShowImage(rclcpp::NodeOptions options, int argc, char ** argv);
```

Both constructors call the execute method, and the latter one does the additional setup.  The first constructor could also make use of ROS parameters for the setup (this can be done in a follow-up PR if you like).

I'm not sure what the best practice is for when there is an error initializing a component. In this case we could throw a runtime error from the constructor, catch it in `main` and exit early.